apps/prisma-sample/src/cat/cat.service.spec.ts

```diff
@@ -0,0 +1,103 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { PrismaService } from '../prisma.service';
+import { CatService } from './cat.service';
+
+const testCat1 = 'Test Cat 1';
+const testBreed1 = 'Test Breed 1';
+
+const catArray = [
+  { name: testCat1, breed: testBreed1, age: 4 },
+  { name: 'Test Cat 2', breed: 'Test Breed 2', age: 3 },
+  { name: 'Test Cat 3', breed: 'Test Breed 3', age: 2 },
+];
+
+const oneCat = catArray[0];
+
+const db = {
+  cat: {
+    findMany: jest.fn().mockResolvedValue(catArray),
+    findUnique: jest.fn().mockResolvedValue(oneCat),
+    findFirst: jest.fn().mockResolvedValue(oneCat),
+    create: jest.fn().mockReturnValue(oneCat),
+    save: jest.fn(),
+    update: jest.fn().mockResolvedValue(oneCat),
+    // as these do not actually use their return values in our sample
+    // we just make sure that their resolve is true to not crash
+    delete: jest.fn().mockResolvedValue(oneCat),
+  },
+};
+
+describe('CatService', () => {
+  let service: CatService;
+  let prisma: PrismaService;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [CatService, PrismaService],
+    })
+      .overrideProvider(PrismaService)
+      .useValue(db)
+      .compile();
+
+    service = module.get<CatService>(CatService);
+    prisma = module.get<PrismaService>(PrismaService);
+  });
```

Comment:
Instead of using `overrideProvider` here, can we set up `PrismaService` as a custom provider?
```suggestion
  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      providers: [
        CatsService,
        {
          provide: PrismaService,
          useValue: db,
      ],
    })
      .compile();

    service = module.get<CatService>(CatService);
    prisma = module.get<PrismaService>(PrismaService);
  });
```